### PR TITLE
Formidable schema test on yml

### PR DESCRIPTION
**Pull request by brunobord**
As you may see, this PR is establishing a set of basic tests on the Form schema, based on the ``formidable.yml`` file that lives in the documentation.
From this perspective, I'm only testing it *as-is*, and I've added a certain number of "FIXME" tags to enhance this definition and fix it to synchronize it with the truth (ie.: the runtime code)

## Review

* [x] Tests
* [x] docs/comments
* [x] `CHANGELOG.rst` Updated
* [ ] Delete your branch

### CODE REVIEW

File: docs/tests/test_fields.py [hunk 1]
```diff
@@ -0,0 +1,340 @@
+from collections import deque
+from copy import deepcopy
+from . import validator, _load_fixture
+
+
+def test_wrong_type_fields_int():
+    form = _load_fixture('0010_wrong_type_field_int.json')
+    errors = sorted(validator.iter_errors(form), key=lambda e: e.path)
```
Comment (syldlb) [inline docs/tests/test_fields.py]: This line occurs very often, have you considered a helper function to improve readability?

File: docs/tests/fixtures/0010_wrong_type_field_dict.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (syldlb) [inline docs/tests/fixtures/0010_wrong_type_field_dict.json]: any reason the number in the file name goes from 0004 to 0010? (also some numbers are similar)

**Comment by brunobord** [comment]
>  Also, I noticed that you have to use pytest in the requirements, how about to switch to it completely?

this one was a bit different. I was just needing pure python tests, and nothing related to Django, so I didn't have to install pytest+pytest-django and try to configure this.

But we may try to open this topic during the next technical grooming.

File: docs/tests/test_fields.py (hunk 1 above)
Comment (brunobord) [reply docs/tests/test_fields.py]: now that you're mentioning it...
I'll see how I can improve this.

File: docs/tests/fixtures/0010_wrong_type_field_dict.json (hunk 2 above)
Comment (brunobord) [reply docs/tests/fixtures/0010_wrong_type_field_dict.json]: I've tried to work using "blocks" of similar tests and "space" them to make sure I could squeeze more tests in the blocks without having to rename all my fixtures.
The numbers are following the schema definition logic, from the most basic to the most complicated.
On top of that, they haven't been written in order, I've jumped from one idea to another, so I've put "spaces" when I was starting to write a test that was a bit ahead, so sometimes there's a gap between fixture numbers.

File: docs/tests/test_fields.py (hunk 1 above)
Comment (brunobord) [reply docs/tests/test_fields.py]: mmmm... thinking about it, I'd be replacing one line of code by... one line of code & a function definition + one line of code... is it worth it?

Comment (syldlb) [reply docs/tests/test_fields.py]: I was thinking this line is a bit "heavy" to read, and replacing it by a function call could make things clearer. But it definitely requires more code... Your call, the PR is approved :+1:

File: docs/tests/fixtures/0010_wrong_type_field_dict.json (hunk 2 above)
Comment (syldlb) [reply docs/tests/fixtures/0010_wrong_type_field_dict.json]: Thanks for the answer :)

**Pull request merged**